src/sentry/static/sentry/app/views/settings/organizationRelay/relayWrapper.tsx

```diff
@@ -114,87 +118,113 @@ class RelayWrapper extends AsyncComponent<Props, State> {
-        >
-          {relays.map(({publicKey: key, name, created, description}) => {
-            const maskedKey = '*************************';
-            return (
-              <React.Fragment key={key}>
-                <Name>
-                  <Text>{name}</Text>
-                  {description && (
-                    <QuestionTooltip position="top" size="sm" title={description} />
-                  )}
-                </Name>
-                <KeyWrapper>
-                  <Key content={maskedKey}>{maskedKey}</Key>
-                  <IconWrapper>
-                    <Clipboard value={key}>
-                      <Tooltip title={t('Click to copy')} containerDisplayMode="flex">
-                        <IconCopy color="gray500" />
-                      </Tooltip>
-                    </Clipboard>
-                  </IconWrapper>
-                </KeyWrapper>
-                <Text>
-                  {!defined(created) ? t('Unknown') : <DateTime date={created} />}
-                </Text>
-                <Actions>
+        {!relays.length ? (
+          <Panel>
+            <EmptyMessage
+              icon={<IconTelescope size="xl" />}
+              title={t('The middle layer between your app and Sentry!')}
+              description={t(
+                'Scrub all personally identifiable information before it arrives in Sentry. Relay improves event response time and acts as a proxy for organizations with restricted HTTP communication.'
+              )}
+              action={
+                <EmptyMessageActions>
+                  <Button href={RELAY_DOCS_LINK} target="_blank">
+                    {t('Go to docs')}
+                  </Button>
                   <Button
-                    size="small"
-                    title={t('Edit Key')}
-                    label={t('Edit Key')}
-                    icon={<IconEdit size="sm" />}
-                    onClick={this.handleOpenEditDialog(key)}
-                  />
-                  <Button
-                    size="small"
-                    title={t('Delete Key')}
-                    label={t('Delete Key')}
-                    onClick={this.handleDelete(key)}
-                    icon={<IconDelete size="sm" />}
-                  />
-                </Actions>
-              </React.Fragment>
-            );
-          })}
-        </StyledPanelTable>
+                    priority="primary"
+                    icon={<IconAdd isCircled />}
+                    onClick={this.handleOpenAddDialog}
+                  >
+                    {t('Register Key')}
+                  </Button>
+                </EmptyMessageActions>
+              }
+            />
+          </Panel>
+        ) : (
+          <StyledPanelTable
+            headers={[t('Display Name'), t('Relay Key'), t('Date Created'), '']}
+          >
+            {relays.map(({publicKey: key, name, created, description}) => {
+              const maskedKey = '*************************';
```

Comment:
Should we move this out of loop/function?

Reply:
sure! I will update it in Part 3, when this component will be refactored.